tcclient/http.go

```diff
@@ -47,6 +48,18 @@ type CallSummary struct {
 	Attempts int
 }
 
+// utility function to create a URL object based on given data
+func setURL(connectionData *ConnectionData, route string, query url.Values) (u *url.URL, err error) {
+	u, err = url.Parse(connectionData.BaseURL + route)
+	if err != nil {
+		return nil, fmt.Errorf("Cannot parse url: '%v', is BaseURL (%v) set correctly?\n%v\n", connectionData.BaseURL+route, connectionData.BaseURL, err)
+	}
+	if query != nil {
+		u.RawQuery = query.Encode()
+	}
+	return
```

Comment:
Yes - if the return types have variables, those variables are initialised on entry into the function/method, and they are returned in the case of a lone `return`. This is another thing I'm doing more, as I think it helps docs to give return types variable names, and also makes code easier to read as when you see that var referenced, you know later it will be returned. I've started doing it - I'm not doing it everywhere, but I prefer it.